### PR TITLE
Developer/nikolaj/add docs

### PR DESCRIPTION
**Pull request by Nikolaj-K**
I wrote an array of docs files or outlines for such, which you may check out here

https://github.com/Constellation-Labs/constellation/tree/developer/nikolaj/add-docs/docs 

Those will then naturally reside under **/constellation/docs**. As the first readme says, those cover:

* Annotated project folder tree
* Architecture diagrams
* Build instructions
* Code base architecture documentation
* Development process
* Discussion of design choices
* Recommended tools and frameworks
* Software dependencies

I fixed some typos in the repo wiki 

https://github.com/Constellation-Labs/constellation/wiki

and, if that PR gets in, will refactor it in line with the files here. I intent to delete the architecture_docs repo, as it's part of this PR here. 
There's now the visual-identity repository (linked below) and its diagram folder will link to that one.

https://github.com/Constellation-Labs/visual-identity

Speaking of **links**, since I wanted to have a lot of crosslinking, some links to this PR will initially refer back to this add-docs branch, but I'll fix them at the latest once those docs propagate up to master (I suppose there's no big hurry).

Some of the files may/should go to the wiki themselves (in particular the `development process` one) and some of the content may/should make it to a .tex (over .md) and then the image folder should go into that tex folder and the .tex would include the images (the diagrams from the archetecture docs).

I'll make a PR for when I list more of the dependencies/tools/frameworks. I'll try to go over any typos before I merge.

### CODE REVIEW

File: docs/desgin-choices.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+This file summarizes current resoning for our approach and design decissions, 
+reviews and references. Please communicate suggestions by making a thread on the 
+[community portal Orion](https://orion.constellationlabs.io/accounts/login/?next=/) 
+or approaching the developers on the
+[discord](https://discordapp.com/invite/KMSmXbV)
+server.
+
+Many decissions can be traced back to the fundamental goal to provide and
+accessible, scalable protocol that focuses on solving the consensus task. 
+For a cursory glipse into the core teams perspectives, you may check out the following clips:
```
Comment (Nikolaj-K) [inline docs/desgin-choices.md]: typo: glimpse

File: README.md [hunk 2]
```diff
@@ -8,106 +7,58 @@ protocol using the
...
-or optionally (to connect to other host)
-```bash
-./build.sh seedhost:port
-```
-3. Interact with app via the API 
-By default this is 
-http://localhost:9000. 
-Look at the file _API.scala_ for endpoints.
+Visit the [/constellation/docs](https://github.com/Constellation-Labs/constellation/docs/) for an explainer of the protocol archetecture or an extended overview of the 
+[project structure](https://github.com/Constellation-Labs/constellation/docs/directory-tree/). Note that the developer branch might be behind the master branch by a few weeks.
```
Comment (tylerprete) [inline README.md]: Do you mean the master branch might be behind the developer branch by a few weeks?

File: docs/desgin-choices.md (hunk 1 above)
Comment (tylerprete) [inline docs/desgin-choices.md]: you mispelled the filename: desgin-choices.md should be design-choices.md

File: docs/desgin-choices.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+* [youtube.com/constellation-labs/talk-at-Tech-Crunch](https://youtu.be/SsYZF4msXuQ) (Aug. 2018, 40 mins)
+* [youtube.com/constellation-labs/testnet-overview](https://youtu.be/xjn6Te7Twg4) (Sept. 2018, 22 mins)
+
+#### on archetecture
```
Comment (tylerprete) [inline docs/desgin-choices.md]: architecture

File: README.md (hunk 2 above)
Comment (Nikolaj-K) [reply README.md]: Yes, lol, good catch.

**Comment by Nikolaj-K** [comment]
I've made some comments to the docs/wiki ticket here

https://github.com/Constellation-Labs/constellation/issues/136#issuecomment-423448174

**Comment by Nikolaj-K** [comment]
In the flow of crosslinking the docs, I refactored the resource list with all the info that has been added since

https://github.com/Constellation-Labs/awesome-constellation

**Comment by Nikolaj-K** [comment]
Created a contributions guidelines page on the wiki, please refactor to your liking

https://github.com/Constellation-Labs/constellation/wiki/contribution-guidelines

**Pull request merged**